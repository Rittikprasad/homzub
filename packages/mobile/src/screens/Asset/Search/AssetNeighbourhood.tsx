/* eslint-disable react/no-unused-state */
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { WithTranslation, withTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { GooglePlacesService } from '@homzhub/common/src/services/GooglePlaces/GooglePlacesService';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { AssetSelectors } from '@homzhub/common/src/modules/asset/selectors';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { CustomMarker } from '@homzhub/common/src/components/atoms/CustomMarker';
import { SelectionPicker } from '@homzhub/common/src/components/atoms/SelectionPicker';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';
import ExploreSections from '@homzhub/mobile/src/components/molecules/ExploreSections';
import { Header } from '@homzhub/mobile/src/components/molecules/Header';
import { Asset } from '@homzhub/common/src/domain/models/Asset';
import { NeighborhoodTabs, PLACES_DATA } from '@homzhub/common/src/constants/AssetNeighbourhood';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { MetricSystems } from '@homzhub/common/src/domain/models/UserPreferences';
import { PlaceTypes } from '@homzhub/common/src/services/GooglePlaces/constants';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/SearchStack';
import { PointOfInterest } from '@homzhub/common/src/services/GooglePlaces/interfaces';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

const EXPANDED = theme.viewport.height * 0.45;

// INTERFACES & TYPES
interface IStateProps {
  asset: Asset | null;
  metricSystem: MetricSystems;
}
interface IOwnState {
  selectedTab: NeighborhoodTabs;
  selectedPlaceType: PlaceTypes;
  isBottomSheetVisible: boolean;
  pointsOfInterest: PointOfInterest[];
  selectedPlaceId: string;
  isApiActive: boolean;
}

type libraryProps = WithTranslation & NavigationScreenProps<SearchStackParamList, ScreensKeys.AssetNeighbourhood>;
type Props = IStateProps & libraryProps;

export class AssetNeighbourhood extends React.Component<Props, IOwnState> {
  private mapRef: MapView | null = null;

  public state = {
    isBottomSheetVisible: false,
    selectedTab: NeighborhoodTabs.Nearby,
    selectedPlaceType: Object.values(PLACES_DATA[NeighborhoodTabs.Nearby])[0].key,
    pointsOfInterest: [],
    selectedPlaceId: '',
    isApiActive: false,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchPOIs();
    this.setState({ isBottomSheetVisible: true });
  };

  public shouldComponentUpdate = (nextProps: Props, nextState: IOwnState): boolean => {
    const { isApiActive, selectedPlaceType, selectedTab } = nextState;
    const { selectedPlaceType: oldType, selectedTab: oldTab } = this.state;
    return !(isApiActive || selectedTab !== oldTab || selectedPlaceType !== oldType);
  };

  public render = (): React.ReactNode => {
    const { asset, t, metricSystem } = this.props;
    const { selectedTab, selectedPlaceType, isBottomSheetVisible, pointsOfInterest, selectedPlaceId } = this.state;

    if (!asset) return null;
    const {
      assetLocation: { longitude, latitude },
    } = asset;
    return (
      <View style={styles.flexOne}>
        <Header type="secondary" icon={icons.leftArrow} onIconPress={this.onBackPress} title={asset.projectName} />
        <MapView
          ref={(mapRef: MapView): void => {
            this.mapRef = mapRef;
          }}
          provider={PROVIDER_GOOGLE}
          style={styles.flexOne}
          initialRegion={{
            latitude,
            longitude,
            latitudeDelta: 0.1,
            longitudeDelta: 0.1,
          }}
        >
          <Marker coordinate={{ latitude, longitude }}>
            <CustomMarker selected />
          </Marker>
          {this.renderMarkers()}
        </MapView>
        <View style={styles.floatingTab}>
          <SelectionPicker
            data={[
              { title: t('nearby'), value: NeighborhoodTabs.Nearby },
              { title: t('commute'), value: NeighborhoodTabs.Commute },
            ]}
            selectedItem={[selectedTab]}
            onValueChange={this.onTabChange}
          />
        </View>
        <BottomSheet
          visible={isBottomSheetVisible}
          renderHeader={false}
          sheetHeight={EXPANDED}
          onCloseSheet={this.onBottomSheetClose}
        >
          <ExploreSections
            placeTypes={Object.values(PLACES_DATA[selectedTab])}
            // @ts-ignore
            selectedPlaceType={PLACES_DATA[selectedTab][selectedPlaceType]}
            onPlaceTypePress={this.onPlaceTypeChange}
            selectedPoiId={selectedPlaceId}
            pointsOfInterest={pointsOfInterest}
            onPoiPress={this.onPoiPress}
            metricSystem={metricSystem}
          />
        </BottomSheet>
      </View>
    );
  };

  private renderMarkers = (): React.ReactNode => {
    const { pointsOfInterest, selectedPlaceId, selectedTab, selectedPlaceType } = this.state;
    // @ts-ignore
    const placeType = PLACES_DATA[selectedTab][selectedPlaceType];

    return pointsOfInterest.map((point: PointOfInterest) => (
      <Marker key={point.placeId} coordinate={{ latitude: point.latitude, longitude: point.longitude }}>
        <Icon
          name={placeType.mapMarker}
          color={selectedPlaceId === point.placeId ? theme.colors.active : theme.colors.darkTint5}
          size={24}
        />
      </Marker>
    ));
  };

  private onTabChange = (selectedTab: NeighborhoodTabs): void => {
    const { selectedTab: oldTab, selectedPlaceType } = this.state;

    let placeTypeToUpdate = selectedPlaceType;
    if (oldTab !== selectedTab) {
      placeTypeToUpdate = Object.values(PLACES_DATA[selectedTab])[0].key;
    }

    this.setState({ selectedTab, selectedPlaceType: placeTypeToUpdate, isBottomSheetVisible: true }, () => {
      this.fetchPOIs().then();
    });
  };

  private onPlaceTypeChange = (selectedPlaceType: PlaceTypes): void => {
    this.setState({ selectedPlaceType }, () => {
      this.fetchPOIs().then();
    });
  };

  private onBottomSheetClose = (): void => {
    this.setState({ isBottomSheetVisible: false });
  };

  private onPoiPress = (poi: PointOfInterest): void => {
    const { latitude, longitude, placeId } = poi;
    this.setState({ selectedPlaceId: placeId });
    this.mapRef?.animateCamera({
      center: {
        latitude,
        longitude,
      },
    });
  };

  private onBackPress = (): void => {
    const {
      navigation: { goBack },
    } = this.props;
    goBack();
  };

  private fetchPOIs = async (): Promise<void> => {
    const { selectedPlaceType } = this.state;
    const { asset, metricSystem } = this.props;

    if (!asset) return;
    this.setState({ isApiActive: true });
    const { assetLocation } = asset;

    try {
      const pointsOfInterest = await GooglePlacesService.getPOIs(
        assetLocation,
        selectedPlaceType,
        undefined,
        metricSystem
      );

      this.setState({ pointsOfInterest, isApiActive: false });
    } catch (e) {
      this.setState({ isBottomSheetVisible: false, isApiActive: false });
      AlertHelper.error({ message: e.message });
    }
  };
}

const styles = StyleSheet.create({
  flexOne: {
    flex: 1,
  },
  floatingTab: {
    position: 'absolute',
    alignItems: 'center',
    left: 0,
    right: 0,
    marginHorizontal: 12,
    top: 120,
  },
});

const mapStateToProps = (state: IState): IStateProps => {
  const { getAsset } = AssetSelectors;
  const { getMetricSystem } = UserSelector;

  return {
    asset: getAsset(state),
    metricSystem: getMetricSystem(state),
  };
};

export default connect(
  mapStateToProps,
  null
)(withTranslation(LocaleConstants.namespacesKey.assetDescription)(AssetNeighbourhood));
