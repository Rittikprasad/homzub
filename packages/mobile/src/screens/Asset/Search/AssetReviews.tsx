import React, { memo, useState, useEffect, useRef } from 'react';
import { StyleSheet, View } from 'react-native';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { AssetRepository } from '@homzhub/common/src/domain/repositories/AssetRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { AssetSelectors } from '@homzhub/common/src/modules/asset/selectors';
import { Dropdown } from '@homzhub/common/src/components/atoms/Dropdown';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { AssetReviewCard } from '@homzhub/mobile/src/components/molecules/AssetReviewCard';
import { AssetReviewsSummary } from '@homzhub/mobile/src/components/molecules/AssetReviewsSummary';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { AssetReview } from '@homzhub/common/src/domain/models/AssetReview';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/SearchStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

const SLIDER_WIDTH = theme.viewport.width - theme.layout.screenPadding * 2;
enum SortType {
  MostRecent = 'MostRecent',
  Positive = 'Positive',
  Critical = 'Critical',
}
type Props = NavigationScreenProps<SearchStackParamList, ScreensKeys.AssetReviews>;

const AssetReviews = (props: Props): React.ReactElement => {
  const {
    navigation,
    route: { params },
  } = props;
  const reviewSummary = useSelector(AssetSelectors.getAssetReviews);
  const { t } = useTranslation(LocaleConstants.namespacesKey.assetDescription);

  const [reviews, setReviews] = useState<AssetReview[]>([]);
  const [sortType, setSortType] = useState<SortType>(SortType.MostRecent);
  const [loading, setLoading] = useState(true);
  const dropDownOptions = useRef([
    { value: SortType.MostRecent, label: t('mostRecent') },
    { value: SortType.Positive, label: t('positive') },
    { value: SortType.Critical, label: t('critical') },
  ]);

  useEffect(() => {
    try {
      AssetRepository.getListingReviews(params).then((response) => {
        setReviews(response.sort((a, b) => new Date(b.modifiedAt).getTime() - new Date(a.modifiedAt).getTime()));
      });
    } catch (err) {
      AlertHelper.error({ message: t('common:genericErrorMessage') });
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    if (reviews.length <= 0) {
      return;
    }

    let sorted: AssetReview[] = [];

    switch (sortType) {
      case SortType.Positive:
        sorted = [...reviews].sort((a, b) => b.rating - a.rating);
        break;
      case SortType.Critical:
        sorted = [...reviews].sort((a, b) => a.rating - b.rating);
        break;
      default:
        sorted = [...reviews].sort((a, b) => new Date(b.modifiedAt).getTime() - new Date(a.modifiedAt).getTime());
        break;
    }

    setReviews(sorted);
  }, [sortType]);

  return (
    <Screen
      backgroundColor={theme.colors.white}
      headerProps={{
        type: 'secondary',
        title: t('customerReviews'),
        onIconPress: navigation.goBack,
      }}
      contentContainerStyle={styles.screen}
      isLoading={loading}
    >
      {reviewSummary && (
        <AssetReviewsSummary reviewSummary={reviewSummary} containerStyle={styles.padding} sliderWidth={SLIDER_WIDTH} />
      )}
      <View style={styles.heading}>
        <Label type="large" textType="semiBold" style={styles.reviewHeading}>
          {t('property:totalReviews', { total: reviews.length })}
        </Label>
        <Dropdown
          data={dropDownOptions.current}
          value={sortType}
          onDonePress={setSortType}
          listHeight={theme.viewport.height / 2}
          testID="drpTimeRange"
          isOutline
          containerStyle={styles.dropdownStyle}
          fontSize="regular"
        />
      </View>
      {reviews.map((review: AssetReview) => (
        <AssetReviewCard key={review.id} review={review} hideButtons />
      ))}
    </Screen>
  );
};

const memoizedComponent = memo(AssetReviews);
export { memoizedComponent as AssetReviews };

const styles = StyleSheet.create({
  screen: {
    paddingHorizontal: 0,
  },
  padding: {
    marginHorizontal: 16,
  },
  heading: {
    marginHorizontal: 16,
    marginVertical: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  reviewHeading: {
    color: theme.colors.darkTint3,
  },
  dropdownStyle: {
    width: 132,
  },
});
