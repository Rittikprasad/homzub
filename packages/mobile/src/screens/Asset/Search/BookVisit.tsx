import React, { Component } from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { CommonActions } from '@homzhub/common/src/modules/common/actions';
import { CommonParamList } from '@homzhub/mobile/src/navigation/Common';
import { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { CollapsibleSection } from '@homzhub/mobile/src/components';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import ScheduleVisitForm from '@homzhub/common/src/components/organisms/ScheduleVisitForm';
import { IReviewRefer } from '@homzhub/common/src/modules/common/interfaces';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

interface IVisitState {
  isCollapsed: boolean;
  isLoading: boolean;
}

interface IDispatchProps {
  setReviewReferData: (payload: IReviewRefer) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<CommonParamList, ScreensKeys.BookVisit>;
type Props = libraryProps & IDispatchProps;

export class BookVisit extends Component<Props, IVisitState> {
  public state = {
    isCollapsed: true,
    isLoading: false,
  };

  public render(): React.ReactNode {
    const { isCollapsed, isLoading } = this.state;
    const {
      t,
      route: { params },
      navigation,
    } = this.props;
    return (
      <Screen
        backgroundColor={theme.colors.white}
        isLoading={isLoading}
        headerProps={{
          onIconPress: navigation.goBack,
          type: 'secondary',
          title: t('assetDescription:BookVisit'),
          icon: icons.close,
        }}
      >
        {/* @ts-ignore */}
        <ScheduleVisitForm
          isCollapsed={isCollapsed}
          isReschedule={params ? params.isReschedule : false}
          leaseListingId={params.lease_listing_id}
          saleListingId={params.sale_listing_id}
          userId={params.userId}
          renderCollapseSection={this.renderCollapsibleSection}
          setLoading={this.setLoading}
          onSubmitSuccess={this.onSuccess}
        />
      </Screen>
    );
  }

  private renderCollapsibleSection = (children: React.ReactElement, title: string): React.ReactElement => {
    const { isCollapsed } = this.state;
    return (
      <CollapsibleSection
        title={title}
        icon={icons.watch}
        titleStyle={styles.upcomingTitle}
        initialCollapsedValue={isCollapsed}
        onCollapse={this.handleSlotView}
      >
        {children}
      </CollapsibleSection>
    );
  };

  private onSuccess = (): void => {
    const { navigation, setReviewReferData } = this.props;
    navigation.goBack();
    setReviewReferData({ isReview: true });
  };

  private handleSlotView = (isCollapsed: boolean): void => {
    this.setState({ isCollapsed });
  };

  private setLoading = (isLoading: boolean): void => {
    this.setState({ isLoading });
  };
}

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { setReviewReferData } = CommonActions;
  return bindActionCreators({ setReviewReferData }, dispatch);
};

const reduxComp = connect(null, mapDispatchToProps)(BookVisit);
export default withTranslation()(reduxComp);

const styles = StyleSheet.create({
  upcomingTitle: {
    marginLeft: 12,
  },
});
