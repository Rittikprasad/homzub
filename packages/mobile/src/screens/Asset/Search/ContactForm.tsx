import React from 'react';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { connect } from 'react-redux';
import { remove, find, cloneDeep } from 'lodash';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { SearchSelector } from '@homzhub/common/src/modules/search/selectors';
import { LeadService } from '@homzhub/common/src/services/LeadService';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { TextArea } from '@homzhub/common/src/components/atoms/TextArea';
import { Avatar } from '@homzhub/common/src/components/molecules/Avatar';
import { RadioButtonGroup } from '@homzhub/common/src/components/molecules/RadioButtonGroup';
import { Header } from '@homzhub/mobile/src/components';
import { MultipleButtonGroup } from '@homzhub/common/src/components/molecules/MultipleButtonGroup';
import { TimeSlotGroup } from '@homzhub/common/src/components/molecules/TimeSlotGroup';
import HandleBack from '@homzhub/mobile/src/navigation/HandleBack';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/SearchStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { IFilter } from '@homzhub/common/src/domain/models/Search';
import { Unit } from '@homzhub/common/src/domain/models/Unit';
import { BedroomType, TimeSlot, UserType } from '@homzhub/common/src/constants/ContactFormData';

interface ISlot {
  from_time: number;
  to_time: number;
}

interface IStateProps {
  filters: IFilter;
  isLoggedIn: boolean;
}

interface IContactState {
  selectedTime: number[];
  selectedSpaces: string[];
  userType: number;
  message: string;
}

type libraryProps = WithTranslation & NavigationScreenProps<SearchStackParamList, ScreensKeys.ContactForm>;
type Props = libraryProps & IStateProps;

export class ContactForm extends React.PureComponent<Props, IContactState> {
  public state = {
    selectedTime: [],
    selectedSpaces: [],
    userType: 1,
    message: '',
  };

  public render = (): React.ReactElement => {
    const { selectedTime, selectedSpaces, userType, message } = this.state;
    const { t, navigation } = this.props;
    return (
      <HandleBack onBack={this.goBack} navigation={navigation}>
        <Header
          icon={icons.close}
          barVisible={false}
          title={t('contact')}
          type="secondary"
          onIconPress={this.goBack}
          testID="header"
        />
        <KeyboardAwareScrollView style={styles.scrollView}>
          {this.renderContactDetail()}
          <Divider containerStyles={styles.divider} />
          <Text type="small" textType="semiBold" style={styles.userType}>
            {t('iAm')}
          </Text>
          <RadioButtonGroup
            data={UserType as Unit[]}
            onToggle={this.onUserToggle}
            containerStyle={styles.radioGroup}
            selectedValue={userType}
          />
          <Label type="large" textType="semiBold" style={styles.label}>
            {t('lookingFor')}
          </Label>
          <MultipleButtonGroup<string>
            data={BedroomType}
            onItemSelect={this.onSelectLookingType}
            selectedItem={selectedSpaces}
            containerStyle={styles.buttonGroup}
          />
          <Label type="large" textType="semiBold" style={styles.label}>
            {t('preferredTime')}
          </Label>
          <TimeSlotGroup data={TimeSlot} selectedItem={selectedTime} onItemSelect={this.onTimeSelect} />
          <TextArea
            label={t('message')}
            placeholder={t('typeYourMessage')}
            value={message}
            containerStyle={styles.textArea}
            onMessageChange={this.handleMessageChange}
          />
          <Button
            type="primary"
            title={t('sendMessage')}
            disabled={selectedTime.length === 0 || selectedSpaces.length === 0}
            icon={icons.envelope}
            iconColor={theme.colors.white}
            iconSize={22}
            titleStyle={styles.buttonTitleStyle}
            containerStyle={styles.buttonStyle}
            onPress={this.handleSubmit}
          />
        </KeyboardAwareScrollView>
      </HandleBack>
    );
  };

  private renderContactDetail = (): React.ReactElement => {
    const {
      t,
      route: {
        params: { contactDetail },
      },
    } = this.props;

    const number = `${contactDetail?.countryCode}${contactDetail?.phoneNumber}`;
    return (
      <>
        <Label type="large" textType="semiBold" style={styles.userType}>
          {t('youContacting')}
        </Label>
        <Avatar
          fullName={contactDetail?.fullName ?? ''}
          designation="Owner"
          phoneNumber={number}
          containerStyle={styles.avatarStyle}
        />
      </>
    );
  };

  private onTimeSelect = (id: number): void => {
    const { selectedTime } = this.state;
    const newTime: number[] = cloneDeep(selectedTime);
    let value;
    if (newTime.includes(id)) {
      remove(newTime, (item: number) => item === id);
      value = newTime;
    } else {
      value = newTime.concat(id);
    }

    this.setState({ selectedTime: value });
  };

  private onUserToggle = (id: number): void => {
    this.setState({
      userType: id,
    });
  };

  private onSelectLookingType = (type: string): void => {
    const { selectedSpaces } = this.state;
    const newSpaces: string[] = cloneDeep(selectedSpaces);
    let value;
    if (newSpaces.includes(type)) {
      remove(newSpaces, (item: string) => item === type);
      value = newSpaces;
    } else {
      value = newSpaces.concat(type);
    }
    this.setState({ selectedSpaces: value });
  };

  private handleMessageChange = (value: string): void => {
    this.setState({ message: value });
  };

  private handleSubmit = async (): Promise<void> => {
    const {
      isLoggedIn,
      filters: { asset_transaction_type },
      route: {
        params: { contactDetail, propertyTermId },
      },
      t,
      navigation,
    } = this.props;
    if (!contactDetail) return;
    const { userType, selectedTime, selectedSpaces, message } = this.state;
    const personType = UserType.find((item) => item.id === userType);
    const timeSlot: ISlot[] = [];

    selectedTime.forEach((id): any => {
      TimeSlot.forEach((item) => {
        if (item.id === id) {
          const slot = {
            from_time: item.from,
            to_time: item.to,
          };
          const slotObject = find(timeSlot, slot);
          if (!slotObject) {
            timeSlot.push(slot);
          }
        }
      });
    });

    if (personType && timeSlot.length > 0) {
      const payload = {
        propertyTermId,
        data: {
          spaces: selectedSpaces,
          contact_person_type: personType.value,
          lead_type: 'MAIL',
          ...(message && { message }),
          person_contacted: contactDetail.id,
          preferred_contact_time: timeSlot,
        },
      };

      if (isLoggedIn && (asset_transaction_type ?? 0) > -1) {
        await LeadService.postLeadDetail(asset_transaction_type ?? 0, payload);
        AlertHelper.success({ message: t('assetDescription:messageSent') });
        navigation.navigate(ScreensKeys.PropertyAssetDescription, { propertyTermId });
      } else {
        AlertHelper.error({ message: t('pleaseSignup') });
      }
    }
  };

  private goBack = (): void => {
    const {
      navigation,
      route: {
        params: { propertyTermId },
      },
    } = this.props;
    navigation.navigate(ScreensKeys.PropertyAssetDescription, { propertyTermId });
  };
}

const mapStateToProps = (state: IState): IStateProps => {
  return {
    filters: SearchSelector.getFilters(state),
    isLoggedIn: UserSelector.isLoggedIn(state),
  };
};

export default connect(mapStateToProps, null)(withTranslation()(ContactForm));

const styles = StyleSheet.create({
  scrollView: {
    paddingHorizontal: 16,
    paddingVertical: 10,
    backgroundColor: theme.colors.white,
  },
  divider: {
    borderColor: theme.colors.darkTint10,
    marginBottom: 24,
  },
  userType: {
    color: theme.colors.darkTint3,
  },
  radioGroup: {
    marginVertical: 16,
  },
  label: {
    marginVertical: 12,
    color: theme.colors.darkTint3,
  },
  buttonGroup: {
    marginBottom: 18,
  },
  textArea: {
    marginVertical: 20,
  },
  buttonTitleStyle: {
    marginHorizontal: 12,
  },
  buttonStyle: {
    flex: 0,
    flexDirection: 'row-reverse',
    margin: 16,
  },
  avatarStyle: {
    paddingVertical: 18,
  },
});
