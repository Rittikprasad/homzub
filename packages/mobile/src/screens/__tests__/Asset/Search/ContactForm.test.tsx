import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ContactForm } from '@homzhub/mobile/src/screens/Asset/Search/ContactForm';
import { SearchFilter } from '@homzhub/common/src/mocks/FilterData';

describe('Contact Form Screen', () => {
  let component: ShallowWrapper;
  let props: any;
  const mock = jest.fn();

  beforeEach(() => {
    props = {
      filters: SearchFilter,
      isLoggedIn: true,
      navigation: {
        goBack: mock,
      },
      route: {
        params: {
          contactDetail: {},
        },
      },
    };
  });

  it('should render snapshot', () => {
    component = shallow(<ContactForm {...props} t={(key: string): string => key} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
