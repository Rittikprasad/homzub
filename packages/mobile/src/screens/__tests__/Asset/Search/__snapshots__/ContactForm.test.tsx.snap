// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form Screen should render snapshot 1`] = `
<HandleBack
  navigation={
    Object {
      "goBack": [MockFunction],
    }
  }
  onBack={[Function]}
>
  <Memo(Header)
    barVisible={false}
    icon="close"
    onIconPress={[Function]}
    testID="header"
    title="contact"
    type="secondary"
  />
  <KeyboardAwareScrollView
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      }
    }
    viewIsInsideTabBar={false}
  >
    <Label
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="large"
    >
      youContacting
    </Label>
    <Memo(Avatar)
      containerStyle={
        Object {
          "paddingVertical": 18,
        }
      }
      designation="Owner"
      fullName=""
      phoneNumber="undefinedundefined"
    />
    <Divider
      containerStyles={
        Object {
          "borderColor": "#E5E8EB",
          "marginBottom": 24,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#33495E",
        }
      }
      textType="semiBold"
      type="small"
    >
      iAm
    </Text>
    <RadioButtonGroup
      containerStyle={
        Object {
          "marginVertical": 16,
        }
      }
      data={
        Array [
          Object {
            "id": 1,
            "label": "Tenant",
            "order": 1,
            "value": "TENANT",
          },
          Object {
            "id": 2,
            "label": "Property Agent",
            "order": 2,
            "value": "PROPERTY_AGENT",
          },
        ]
      }
      onToggle={[Function]}
      selectedValue={1}
    />
    <Label
      style={
        Object {
          "color": "#33495E",
          "marginVertical": 12,
        }
      }
      textType="semiBold"
      type="large"
    >
      lookingFor
    </Label>
    <MultipleButtonGroup
      containerStyle={
        Object {
          "marginBottom": 18,
        }
      }
      data={
        Array [
          Object {
            "title": "1BHK",
            "value": "1BHK",
          },
          Object {
            "title": "2BHK",
            "value": "2BHK",
          },
          Object {
            "title": "3BHK",
            "value": "3BHK",
          },
          Object {
            "title": "4BHK",
            "value": "4BHK",
          },
          Object {
            "title": "5BHK+",
            "value": "5BHK+",
          },
        ]
      }
      onItemSelect={[Function]}
      selectedItem={Array []}
    />
    <Label
      style={
        Object {
          "color": "#33495E",
          "marginVertical": 12,
        }
      }
      textType="semiBold"
      type="large"
    >
      preferredTime
    </Label>
    <TimeSlotGroup
      data={
        Array [
          Object {
            "formatted": "8 AM - 12 PM",
            "from": 8,
            "icon": "sun-rise",
            "id": 1,
            "to": 12,
          },
          Object {
            "formatted": "12 PM - 3 PM",
            "from": 12,
            "icon": "sun",
            "id": 2,
            "to": 15,
          },
          Object {
            "formatted": "3 PM - 6 PM",
            "from": 15,
            "icon": "sun-set",
            "id": 3,
            "to": 18,
          },
          Object {
            "formatted": "6 PM - 9 PM",
            "from": 18,
            "icon": "moon",
            "id": 4,
            "to": 21,
          },
        ]
      }
      onItemSelect={[Function]}
      selectedItem={Array []}
    />
    <TextArea
      containerStyle={
        Object {
          "marginVertical": 20,
        }
      }
      label="message"
      onMessageChange={[Function]}
      placeholder="typeYourMessage"
      value=""
    />
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "flexDirection": "row-reverse",
          "margin": 16,
        }
      }
      disabled={true}
      icon="envelope"
      iconColor="#FFFFFF"
      iconSize={22}
      onPress={[Function]}
      title="sendMessage"
      titleStyle={
        Object {
          "marginHorizontal": 12,
        }
      }
      type="primary"
    />
  </KeyboardAwareScrollView>
</HandleBack>
`;
